tests/src/Kernel/CivicrmEntityTestBase.php

```diff
@@ -27,33 +31,105 @@ abstract class CivicrmEntityTestBase extends KernelTestBase {
     'datetime',
   ];
 
+  public function alter(ContainerBuilder $container) {
+    $this->mockCiviCrmApi($container);
+  }
+
   /**
    * {@inheritdoc}
    */
   protected function setUp() {
     parent::setUp();
-    require __DIR__ . '/../Type.php';
-    require __DIR__ . '/../CiviCrmDaoStubs.php';
-    $this->mockCiviCrmApi();
+    define('CIVICRM_CONTAINER_CACHE', 'never');
+    define('CIVICRM_TEST', 'never');
+
+    // Add file_private_path, required to properly set templateCompilePath.
+    $file_private_path = $this->siteDirectory . 'files/private';
+    $this->setSetting('file_private_path', $file_private_path);
+    mkdir($file_private_path, 0775);
+
+    module_load_install('civicrm');
+    $setup = _civicrm_setup();
+    $installed = $setup->checkInstalled();
+    if ($installed->isSettingInstalled() || $installed->isDatabaseInstalled()) {
+      throw new \Exception("CiviCRM appears to have already been installed. Skipping full installation.");
+    }
+
+    \Civi\Setup::dispatcher()
+      ->addListener('civi.setup.installFiles', function (\Civi\Setup\Event\InstallFilesEvent $e) use ($file_private_path) {
+        $model = $e->getModel();
+        $model->settingsPath = implode(DIRECTORY_SEPARATOR, [$this->siteDirectory, 'civicrm.settings.php']);
+        $model->templateCompilePath = implode(DIRECTORY_SEPARATOR, [$file_private_path, 'civicrm', 'templates_c']);
+      }, 900);
+
```

Comment:
See https://lab.civicrm.org/dev/core/-/issues/2590. This works around the fact the path is `vfs://` and CiviCRM prepends the root path to Drupal.